File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/PlayerDirectory.java
package top.likoslupus.cellulosesz.api.player;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

/**
 * Server-thread online-player snapshots.
 */
public interface PlayerDirectory {

    List<CellPlayer> onlinePlayers();

    Optional<CellPlayer> onlinePlayer(UUID uuid);

    Optional<CellPlayer> onlinePlayer(String name);

    List<String> onlinePlayerNames();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/PlayerDirectory.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*

/**
 * Server-thread online-player snapshots.
 */
public interface PlayerDirectory {

    public fun onlinePlayers(): List<CellPlayer>

    public fun onlinePlayer(uuid: UUID): CellPlayer?

    public fun onlinePlayer(name: String): CellPlayer?

    public fun onlinePlayerNames(): List<String>

}
